hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallerContext.java

```diff
@@ -54,7 +58,7 @@
   private final byte[] signature;
 
   private CallerContext(Builder builder) {
-    this.context = builder.context;
+    this.context = builder.sb.length() > 0 ? builder.sb.toString() : null;
```

Comment:
I think it is cleaner to make it a method: getContext() in the Builder so we do the checks there.

Reply:
This makes sense. It's clearer.